### DISCUSSION TITLE
Dynamic Linking Slint in Dev.Mode / reduce dependencies

### DISCUSSION THREAD

**Discussion by renlite** [slint-ui/slint#5740] [General] [answered]
1)
Is it possible to use a Dynamic Library of Slint to develope with Rust?
e.g.: $ cargo run --features slint/dynamic_linking

2)
First Slint-App build downloads about 350 crates. How can I reduce the used dependencies? There are i-slint* crates that are not always used (e.g.: i-slint-backend-qt).
- What ist the min. config in cargeo.toml of dependencies for rendering with femtovg?
- Can I exclude some crates in Rust-Analizer, to avoid indexing of all the modules?

**Comment by tronical** [comment, marked as answer]
> Is it possible to use a Dynamic Library of Slint to develope with Rust? e.g.: $ cargo run --features slint/dynamic_linking

This is not implemented at the moment. I believe bevy has such a method to speed up linking. I suppose something similar could be implemented for Slint, perhaps. 
 
> First Slint-App build downloads about 350 crates. How can I reduce the used dependencies? There are i-slint* crates that are not always used (e.g.: i-slint-backend-qt).
> 
> * What ist the min. config in cargeo.toml of dependencies for rendering with femtovg?

The minimum would be `default-features = false, features = ["backend-winit", "renderer-femtovg", "compat-1-2"]` on a desktop or embedded linux system.

> * Can I exclude some crates in Rust-Analizer, to avoid indexing of all the modules?

I don't know if that's possible, but it sounds similar to https://github.com/rust-lang/rust-analyzer/issues/8275
